### DISCUSSION TITLE
Query to fetch tokens by account issues

### DISCUSSION THREAD

**Discussion by CryptainGolD** [aptos-labs/aptos-developer-discussions#537] [Questions] [answered]
```ts
query GetAccountNfts($address: String) {
current_token_ownerships_v2(
  where: {owner_address: {_eq: $address}, amount: {_gt: "0"}}
) {
  current_token_data {
    collection_id
    largest_property_version_v1
    current_collection {
      collection_id
      collection_name
      description
      creator_address
      uri
      __typename
    }
    description
    token_name
    token_data_id
    token_standard
    token_uri
    __typename
  }
  owner_address
  amount
  __typename
}
}
```

This query fetches tokens by account, but cannot fetch listed tokens.
Like the Petra wallet. it can't show listed tokens. But Rarible marketplace shows all tokens even listed.
How can I fetch even listed tokens with query?

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by 0xbrw** [comment, marked as answer]
Once your tokens are listed on a marketplace, the ability to view them depends on the marketplace’s functionality.

If you need to query token balances, you can use the interactive indexer in our docs: [Fungible Asset Balances](https://aptos.dev/en/build/indexer/fungible-asset-balances).
